### ISSUE TITLE
Handling of {multline*} equations

### ISSUE THREAD

**Issue by dginev** [brucemiller/LaTeXML#887]
I think I hit an odd edge case while developing with an amsmath mulline equation. A minimal example would be:
```tex
\begin{multline*}
a \\
b \\
c \\
\end{multline*}
```

The curious bit is that converting this equation to XML produces tex/alttext attributes that only preserve the information of `\displaystyle` but "forget" the multiline aspect of the equation. I am now getting a strange half-memory of someone warning me about such advanced cases at some point in the past, and another half-memory of me not worrying too much about it back then.

Well, then was then, but now I am really interested in producing good SVG from these equations (until MathML makes a dashing appearance on Chrome), so I need to preserve that information into the tex attribute somehow, if the SVG conversion is to produce the correct output.

I wonder if that is too much of a tall order? A "hacky" alternative would be to take a "peek" while doing `extractTeX` for MathImages and check if there are any `\\` terminators - at which point we can safely use the multiline wrapper instead of the `\beginDISPLAY` one.

@brucemiller what sounds most reasonable? I guess this is a relatively rare scenario as most multi line equation (groups) get broken up into html chunk, and the math snippets that need to become SVG are single-line fragments, right?

**Comment by brucemiller** [comment]
hmm,odd; the `\begin{multline}` stuff is in the reversion, but apparently not taking effect. Have to dig.

**Comment by brucemiller** [comment]
Grr... it's because the reversion code is explicitly only reverting the body, expecting the environment to be equation (or equivalent to).  Ugh...

**Comment by dginev** [comment]
Yes, it certainly looked that way to me. Hence seemed like it needed an upgrade. Not the most pressing of issues, but seems necessary if the SVG math output is to work reliably for different math environments.

**Comment by dginev** [comment]
[0.8.4 note] Pushing milestone back on non-urgent issues.

**Comment by dginev** [comment]
I took a debug dive here and can report the responsible pieces, since I can't quite see the full path to the patch just yet (still learning about multline and reversions).

1. The reversion for a `ltx:Math` element adds the reversion of a Whatsit's body, when available via:
    ```perl
    Tag('ltx:Math', afterClose => \&add_body_TeX);
    ```
2. The multline and multline* bindings end up with bodies *and* redundant first arguments for their Whatsits, via an admittedly hacky trick:
    ```perl
      afterDigest => sub {    # Nasty trick required to make "body" be the arg.
    $_[1]->setProperty('MULTIROW_ALIGNMENT_RULE' => { '0' => 'left', '-1' => 'right' });
    $_[1]->setBody($_[1]->getArg(1)->unlist, undef); },
    ```
3. As to the main reversion, I don't think it is ever used. It was deposited at `DefConstructor('\@@multline` as:
    ```perl
    reversion      => '\begin{multline}#1\end{multline}',
    ```
4. And in fact if you put some strategic prints you see that it is in fact filled in, and wrongly as '\begin{multline}#1\end{multline}#body', as the argument and body are the same and currently both present.

So maybe the right summary here is that the binding definitions for `{multline}` and `{multline*}` need to be reworked, and likely other equation environments could also benefit from getting their reversions examined...

A simple idea that comes to mind is to add special macros that expand to an empty Tokens() but have a reversion to the `\begin{multline}` and `\end{multline}`, and then strategically add them to the `->setBody` call I mentioned above. Though that feels like just expanding the hackiness... Can be solved in the next release, just wanted to take a fresh look before shelving it again.

**Comment by brucemiller** [comment]
It also deserves mention the issue of how the `tex` attribute is expected to be used. It was designed to support conversion of the math to an image. The idea is to embed the math content into a special "environment", analogous to either `$...$` or `\[...\]` (for inline & display), but providing the ability, when applying a regular tex/latex to it, to capture & measure the resulting math, as well as dump it onto a separate page to be snipped out.  Getting arbitrary TeX is tricky: what if it wants to place equation numbers, how do we measure the box it created, etc.

So, we `add_body_TeX` needs to back off from avoiding the containing environment, which should atomatically pick up the multiline, but redesign the backend to be able to accept arbitrary environments there. The latter part seems the trickiest.
